src/libraries/System.Private.CoreLib/src/System/Random.cs

```diff
@@ -155,14 +183,36 @@ public virtual void NextBytes(byte[] buffer)
 
         /// <summary>Returns a random floating-point number between 0.0 and 1.0.</summary>
         /// <returns>A double-precision floating point number that is greater than or equal to 0.0, and less than 1.0.</returns>
-        protected virtual double Sample() => _impl.Sample();
+        protected virtual double Sample()
+        {
+            double result = _impl.Sample();
+            AssertInRange(result);
+            return result;
+        }
 
         private static void ThrowMaxValueMustBeNonNegative() =>
             throw new ArgumentOutOfRangeException("maxValue", SR.Format(SR.ArgumentOutOfRange_NeedNonNegNum, "maxValue"));
 
         private static void ThrowMinMaxValueSwapped() =>
             throw new ArgumentOutOfRangeException("minValue", SR.Format(SR.Argument_MinMaxValue, "minValue", "maxValue"));
 
+        [Conditional("DEBUG")]
```

Comment:
One would hope the jit would see that all the code inside this method is debug only and this annotation would be superfluous

Reply:
With the attribute it's removed by the C# compiler long before it gets to the jit :)